src/Service.GraphQLBuilder/GraphQLStoredProcedureBuilder.cs

```diff
@@ -101,6 +201,30 @@ public static List<JsonDocument> FormatStoredProcedureResultAsJsonList(JsonDocum
             return resultJson;
         }
 
+        /// <summary>
+        /// Takes the result from DB as JsonDocument and formats it in a way that can be filtered by column
+        /// name. It parses the Json document into a list of Dictionary with key as result_column_name
+        /// with it's corresponding value.
+        /// returns an empty list in case of no result 
+        /// or stored-procedure is trying to read from DB without READ permission.
+        /// </summary>
+        public static List<JsonDocument> FormatStoredProcedureObjectResultAsJsonList(JsonDocument? jsonDocument)
+        {
+            if (jsonDocument is null)
+            {
+                return new List<JsonDocument>();
+            }
+
+            List<JsonDocument> resultJson = new();
```

Comment:
We've been in the process of refactoring away heavy usage of JsonDocument as a return type and instead returning the JsonElement acquired by calling JsonDocument.Root.Clone(). An example/explanation can be found here: https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/use-dom-utf8jsonreader-utf8jsonwriter?pivots=dotnet-6-0#jsondocument-is-idisposable